test/e2e-secondary-network/framework.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	aggregatorclientset "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+
+	crdclientset "antrea.io/antrea/pkg/client/clientset/versioned"
+	antreae2e "antrea.io/antrea/test/e2e"
+)
+
+type SNTestData struct {
+	e2eTestData        *antreae2e.TestData
+	kubeConfig         *restclient.Config
+	clientset          kubernetes.Interface
+	aggregatorClient   aggregatorclientset.Interface
+	crdClient          crdclientset.Interface
+	logsDirForTestCase string
+	testNamespace      string
+}
+
+func createDirectory(path string) error {
+	return os.Mkdir(path, 0700)
+}
+
+const (
+	busyboxImage = "projects.registry.vmware.com/antrea/busybox"
+)
+
+var SNtestData *SNTestData
```

Comment:
I would just rename it to `testData` / `TestData`, there is no risk of name collision since this is in its own package